### PR TITLE
Validates Book fields before adding/saving

### PR DESCRIPTION
**Pull request by naviza**
We can add isbn validator too, if necessary
Screenshot_20201027-110601_Bookworm_Adventure_Deluxe_2

### CODE REVIEW

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java [hunk 1]
```diff
@@ -105,6 +107,36 @@ public void saveBook(View view)
...
+            EditTextValidator.isEmpty(titleView);
+            valid = false;
+        }
+        if (TextUtils.isEmpty(authorView.getText().toString()))
+        {
+            EditTextValidator.isEmpty(authorView);
+            valid = false;
+        }
+        /*
+        // Leaving this in in case the requirements include the ISBN to be non-empty
```
Comment (huddy987) [inline app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java:130]: I'm unsure on the requirement for this, but maybe we could make it for now either be allowed to be empty or the length of however many digits the isbn should be? So you could either leave it empty or n digits? Then later we can remove the empty part if we find out that isbn should always be included

**Review by huddy987 — CHANGES_REQUESTED** [review]

**Review by huddy987 — APPROVED** [review]

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java [hunk 2]
```diff
@@ -105,6 +107,40 @@ public void saveBook(View view)
...
+            valid = false;
+        }
+        if (TextUtils.isEmpty(authorView.getText().toString()))
+        {
+            EditTextValidator.isEmpty(authorView);
+            valid = false;
+        }
+        // ISBN can be empty, or has digits of length 10 or 13
+        // https://en.wikipedia.org/wiki/International_Standard_Book_Number
+        String isbn_input = isbnView.getText().toString();
```
Comment (KateVlaar) [inline app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java:132]: Are we sure we want to allow this to be empty? Later down the line we might be able to compare it to scans to make sure users are handing over the right book

Comment (huddy987) [reply app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java:132]: We should ask the TA if isbn is mandatory

Comment (KateVlaar) [reply app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java:132]: Might as well ask about description too while we're at it

**Review by KateVlaar — APPROVED** [review]

**Pull request merged**